org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/AbstractNotificationPopup.java

```diff
@@ -0,0 +1,567 @@
+
+  private static final int MAX_WIDTH = 400;
+
+  private static final int MIN_HEIGHT = 100;
+
+  private static final long DEFAULT_DELAY_CLOSE = 8 * 1000;
+
+  private static final int PADDING_EDGE = 5;
+
+  private long delayClose = DEFAULT_DELAY_CLOSE;
+
+  protected LocalResourceManager resources;
+
+  private GradientColors color;
+
+  private final Display display;
+
+  private Shell shell;
+
+  private Region lastUsedRegion;
+
+  private Image lastUsedBgImage;
+
+  private final Job closeJob = new Job(LABEL_JOB_CLOSE) {
+
+    @Override
+    protected IStatus run(IProgressMonitor monitor) {
+      if (!display.isDisposed()) {
+        display.asyncExec(new Runnable() {
+          public void run() {
+            Shell shell = AbstractNotificationPopup.this.getShell();
+            if (shell == null || shell.isDisposed()) {
+              return;
+            }
+
+            if (isMouseOver(shell)) {
+              scheduleAutoClose();
+              return;
+            }
+
+            AbstractNotificationPopup.this.closeFade();
+          }
+
+        });
+      }
+      if (monitor.isCanceled()) {
+        return Status.CANCEL_STATUS;
+      }
+
+      return Status.OK_STATUS;
+    }
+  };
+
+  private final boolean respectDisplayBounds = true;
+
+  private final boolean respectMonitorBounds = true;
+
+  private FadeJob fadeJob;
+
+  private boolean fadingEnabled;
+
+  public AbstractNotificationPopup(Display display) {
+    this(display, SWT.NO_TRIM | SWT.ON_TOP | SWT.NO_FOCUS | SWT.TOOL);
+  }
+
+  public AbstractNotificationPopup(Display display, int style) {
+    super(new Shell(display));
+    setShellStyle(style);
+
+    this.display = display;
+    resources = new LocalResourceManager(JFaceResources.getResources());
+    initResources();
+
+    closeJob.setSystem(true);
+  }
+
+  public boolean isFadingEnabled() {
+    return fadingEnabled;
+  }
+
+  public void setFadingEnabled(boolean fadingEnabled) {
+    this.fadingEnabled = fadingEnabled;
+  }
+
+  protected abstract String getPopupShellTitle();
+
+  protected Image getPopupShellImage(int maximumHeight) {
+    return null;
+  }
+
+  /**
+   * Override to populate with notifications.
+   * 
+   * @param parent
+   */
+  protected void createContentArea(Composite parent) {
+    // empty by default
+  }
+
+  /**
+   * Override to customize the title bar
+   */
+  protected void createTitleArea(Composite parent) {
+    ((GridData) parent.getLayoutData()).heightHint = TITLE_HEIGHT;
+
+    Label titleImageLabel = new Label(parent, SWT.NONE);
+    titleImageLabel.setImage(getPopupShellImage(TITLE_HEIGHT));
+
+    Label titleTextLabel = new Label(parent, SWT.NONE);
+    titleTextLabel.setText(getPopupShellTitle());
+    titleTextLabel.setFont(JFaceResources.getFontRegistry().getBold(JFaceResources.DEFAULT_FONT));
+    titleTextLabel.setForeground(getTitleForeground());
+    // titleTextLabel.setForeground(color.getTitleText());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)